presto-parser/src/main/java/com/facebook/presto/sql/tree/Window.java

```diff
@@ -13,99 +13,21 @@
  */
 package com.facebook.presto.sql.tree;
 
-import com.google.common.collect.ImmutableList;
-
-import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
 
-import static com.google.common.base.MoreObjects.toStringHelper;
-import static java.util.Objects.requireNonNull;
-
-public class Window
+public abstract class Window
         extends Node
 {
-    private final List<Expression> partitionBy;
-    private final Optional<OrderBy> orderBy;
-    private final Optional<WindowFrame> frame;
-
-    public Window(List<Expression> partitionBy, Optional<OrderBy> orderBy, Optional<WindowFrame> frame)
-    {
-        this(Optional.empty(), partitionBy, orderBy, frame);
-    }
-
-    public Window(NodeLocation location, List<Expression> partitionBy, Optional<OrderBy> orderBy, Optional<WindowFrame> frame)
-    {
-        this(Optional.of(location), partitionBy, orderBy, frame);
-    }
-
-    private Window(Optional<NodeLocation> location, List<Expression> partitionBy, Optional<OrderBy> orderBy, Optional<WindowFrame> frame)
+    public Window(Optional<NodeLocation> location)
     {
         super(location);
-        this.partitionBy = requireNonNull(partitionBy, "partitionBy is null");
-        this.orderBy = requireNonNull(orderBy, "orderBy is null");
-        this.frame = requireNonNull(frame, "frame is null");
     }
 
-    public List<Expression> getPartitionBy()
-    {
-        return partitionBy;
-    }
-
-    public Optional<OrderBy> getOrderBy()
-    {
-        return orderBy;
-    }
-
-    public Optional<WindowFrame> getFrame()
-    {
-        return frame;
-    }
+    public abstract Optional<WindowFrame> getFrame();
```

Comment:
This does not make sense because only one implementation of this class returns something useful here.